File: 1302-delete-characters-to-make-fancy-string/1302-delete-characters-to-make-fancy-string.java
class Solution {
    public String makeFancyString(String s) {
        StringBuilder res = new StringBuilder();
        int count = 1; // Start count at 1 for the first character
        res.append(s.charAt(0));
        
        for (int i = 1; i < s.length(); i++) {
            if (s.charAt(i) == s.charAt(i - 1)) {
                count++;
            } else {
                count = 1; // Reset count for a new character
            }
            
            if (count <= 2) {
                res.append(s.charAt(i));
            }
        }
        
        return res.toString();
    }
}

File: 1302-delete-characters-to-make-fancy-string/1302-delete-characters-to-make-fancy-string.py
class Solution(object):
    def makeFancyString(self, s):
        """
        :type s: str
        :rtype: str
        """
        res = ""

        i = 0
        
        while i < len(s):
            res += s[i]
            if i<len(s)-1 and s[i] == s[i+1]:
                j = i+1
                while j<len(s) and s[j] == s[i]:
                    j+=1
                i = j
                res+=s[j-1]
            else:
                i+=1
        
        return res